extractor/extraction_containers.cpp

```diff
@@ -479,6 +484,8 @@ void ExtractionContainers::WriteEdges(std::ofstream& file_out_stream) const
             continue;
         }
 
+        // IMPORTANT!!!!
+        // Here, we depend on slicing to write out the base class (not NodeBasedEdgeWithOSM)
         file_out_stream.write((char*) &edge.result, sizeof(NodeBasedEdge));
         number_of_used_edges++;
```

Comment:
Hmm slicing is something different. Slicing is when you have `base` and `child`, and instead of binding a child to a base with references or pointers for polymorphism like `base& myBase = myChild;` you accidentally say `base b = myChild;`, then `b` really is a `base` slicing off its `child` part.

What you are doing here is just writing out the first n bytes of the base, hoping the memory layout is base-first-then-child. I haven't looked deeper into it yet, but I can show you at least a counterexample, of when this breaks:

https://gist.github.com/daniel-j-h/71e33b698234e3deada2#file-virtual-txt-L33

I also think you can guard against this via static asserting on [std::is_polymorphic](http://en.cppreference.com/w/cpp/types/is_polymorphic).

It's probably a good idea to encapsulate and hide all of this.